gazelle/config.go

```
@@ -0,0 +1,253 @@
+// the directive is specified multiple times in the Bazel workspace.
+func (c *TypeScriptConfig) SetValidateImportStatements(validate bool) {
+	c.validateImportStatements = validate
+}
+
+// ValidateImportStatements returns whether the TypeScript import statements should
+// be validated or not. If this option was not explicitly specified by the user,
+// it defaults to true.
+func (c *TypeScriptConfig) ValidateImportStatements() bool {
+	return c.validateImportStatements
+}
+
+// SetLibraryNamingConvention sets the ts_project target naming convention.
+func (c *TypeScriptConfig) SetLibraryNamingConvention(libraryNamingConvention string) {
+	c.libraryNamingConvention = libraryNamingConvention
+}
+
+// RenderLibraryName returns the ts_project target name by performing all
+// substitutions.
+func (c *TypeScriptConfig) RenderLibraryName(packageName string) string {
+	return strings.ReplaceAll(c.libraryNamingConvention, packageNameNamingConventionSubstitution, packageName)
+}
+
+// SetTestsNamingLibraryConvention sets the ts_project test target naming convention.
+func (c *TypeScriptConfig) SetTestsNamingLibraryConvention(testsNamingConvention string) {
+	c.testsNamingConvention = testsNamingConvention
+}
+
+// RenderTestName returns the ts_project test target name by performing all
+// substitutions.
+func (c *TypeScriptConfig) RenderTestsLibraryName(packageName string) string {
+	return strings.ReplaceAll(c.testsNamingConvention, packageNameNamingConventionSubstitution, packageName)
+}
+
+func (c *TypeScriptConfig) SetSourceFileGlob(srcsFileGlob string) {
+	c.srcsFileGlob = srcsFileGlob
+}
+func (c *TypeScriptConfig) IsSourceFile(filePath string) bool {
+	if !isSourceFileType(filePath) {
+		return false
+	}
+	if c.srcsFileGlob == "" {
+		return true
+	}
+
+	m, e := doublestar.Match(c.srcsFileGlob, filePath)
+
+	if e != nil {
+		fmt.Println("ERROR: ", fmt.Errorf("srcs file glob error %e", e))
```

```suggestion
	m, err := doublestar.Match(c.srcsFileGlob, filePath)
	if err != nil {
		fmt.Printf("ERROR: srcs file glob error: %v\n", err)
```

---

Interesting that you had `%e` for error, but it's meant for scientific notation of floating-point numbers.
https://pkg.go.dev/fmt

---

Also, this may beg a bigger question: do you care about the error at all? From the `Match` comment:
```
// Match requires pattern to match all of name, not just a substring.
// The only possible returned error is ErrBadPattern, when pattern
// is malformed.
```

---

> Also, this may beg a bigger question: do you care about the error at all? From the Match comment:

This means the user (developer) set an invalid directive so I think outputting an error and failing is a clear indication to the user. Where simply ignoring the error would add more files then necessary to the BUILD and could easily go unnoticed by the dev?